### PR TITLE
Restore previously broken serialization due to parameter integration.

### PR DESCRIPTION
**Pull request by karandesai-96**
Mostly this is chore work, removal of old redundant methods.

### CODE REVIEW

File: test/test_convolutional_layer.h [hunk 1]
```diff
@@ -25,7 +25,7 @@ TEST(convolutional, setup_internal) {
   EXPECT_EQ(l.out_data_size(), 18u);             // size of output tensors
   EXPECT_EQ(l.fan_in_size(), 9u);                // num of incoming connections
   EXPECT_EQ(l.fan_out_size(), 18u);              // num of outgoing connections
-  EXPECT_EQ(l.weights_at()[0]->size(), 18u);     // size of weights
+  EXPECT_EQ(l.weights_at()[0]->size(), 36u);     // size of weights
```
Comment (nyanp) [inline test/test_convolutional_layer.h]: @karandesai-96 Is it correct? Size of shape should be 3x3x1x2=18.

File: test/test_deconvolutional_layer.h [hunk 2]
```diff
@@ -33,7 +33,7 @@ TEST(deconvolutional, setup_tiny) {
   EXPECT_EQ(l.fan_in_size(), 9u);             // num of incoming connections
   EXPECT_EQ(l.fan_out_size(), 18u);           // num of outgoing connections
   EXPECT_EQ(l.parameters().size(), 2u);       // num of trainable parameters
-  EXPECT_EQ(l.weights_at()[0]->size(), 18u);  // size of weight parameter
+  EXPECT_EQ(l.weights_at()[0]->size(), 36u);  // size of weight parameter
```
Comment (nyanp) [inline test/test_deconvolutional_layer.h]: same as convolutional_layer

File: test/test_network.h [hunk 3]
```diff
@@ -137,17 +137,17 @@ TEST(network, manual_init) {
...
+  EXPECT_EQ(c1_w.size(), 9u);
+  EXPECT_EQ(c1_b.size(), 1u);
+  EXPECT_EQ(f1_w.size(), 2u);
 
-  *c1_w = {0, 1, 2, 3, 4, 5, 6, 7, 8};
-  *c1_b = {1};
-  *f1_w = {1, 2};
+  c1_w.set_data(Tensor<float_t>{vec_t{0, 1, 2, 3, 4, 5, 6, 7, 8}});
+  c1_b.set_data(Tensor<float_t>{vec_t{1}});
+  f1_w.set_data(Tensor<float_t>{vec_t{1, 2}});
```
Comment (nyanp) [inline test/test_network.h]: Consider adding initializer-list assignment operator to ```Parameter``` to simplify these codes.

File: test/test_serialization.h [hunk 4]
```diff
@@ -46,8 +46,11 @@ TEST(serialization, serialize_avepool) {
                     "height": 10,
                     "depth": 3
                 },
-                "pool_size_x": 2,
-                "pool_size_y": 2,
+                "window": {
+                    "width": 2,
+                    "height": 2,
+                    "depth": 3
+                },
```
Comment (nyanp) [inline test/test_serialization.h]: We can keep it as is. Do we need to break backward compatibility?

File: test/test_network.h (hunk 3 above)
Comment (Randl) [inline test/test_network.h]: Also, you don't need to write `float_t`, `Tensor<>` is enough and much shorter

**Comment by karandesai-96** [comment]
Coverage restored! (+10%) :smile:

File: test/test_network.h (hunk 3 above)
Comment (karandesai-96) [inline test/test_network.h]: @nyanp you're right!
@Randl yes, thanks!

File: test/test_serialization.h (hunk 4 above)
Comment (karandesai-96) [inline test/test_serialization.h]: Changed under the hood but user facing API is kept backward compatible now.

**Pull request merged**